core/modules/layout/layout.admin.inc

```diff
@@ -937,6 +937,9 @@ function layout_content_form_submit($form, &$form_state) {
   // Delete the tempstore copy of the layout.
   layout_clear_layout_tempstore($layout->name);
 
+  // Reset the block usage report.
+  backdrop_static_reset('layout_get_block_usage');
```

Comment:
This isn't necessary IMO, there's no way (or at least unlikely) that you would be on the layout_content_form() and the Layout report page within a single page load.